pkg/controller/v1beta1/inferenceservice/components/transformer.go

```diff
@@ -79,6 +87,32 @@ func (p *Transformer) Reconcile(isvc *v1beta1.InferenceService) error {
 		}),
 		Annotations: annotations,
 	}
+
+	// Need to wait for predictor URL in modelmesh deployment mode
+	if isvcutils.GetDeploymentMode(annotations, deployConfig) == constants.ModelMeshDeployment {
+		// check if predictor URL is populated
+		if isvc.Status.Components["predictor"].URL == nil {
+			// exit transformer reconcile with an error when predictor URL not populated
+			return fmt.Errorf("Predictor URL from ModelMesh is not ready")
```

Comment:
I think we would want to differentiate this from actual errors since it just means "not ready yet". Instead of returning an error from the top-level reconcile func it would be better to return e.g. `ctrl.Result{RequeueAfter: 2 * time.Second}` (and could log an info message here).

I guess an additional return value might need to be added to this func to facilitate that.

Reply:
@njhill This requires interface changes as well. I will be happy to make such changes if necessary.

Reply:
Thanks @chinhuang007, looks like you already made the change.